sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +20,226 @@
 s3 = boto3.client("s3")
 
 
+def http_response(headers=None, status_code=200, body="Success"):
+    """Customizes HTTP response for the lambda function.
+
+    Parameters
+    ----------
+    headers : dict, optional
+        Content headers for the response, defaults to Content-type: text/html.
+    status_code : int, optional
+        HTTP status code indicating the result of the operation, defaults to 200.
+    body : str, optional
+        The content of the response, defaults to 'Success'.
+
+    Returns
+    -------
+    dict
+        A dictionary containing headers, status code, and body, designed to be returned
+        by a Lambda function as an API response.
+    """
+    if headers is None:
+        headers = (
+            {
+                "Content-Type": "text/html",
+            },
+        )
+    return {
+        "headers": headers,
+        "statusCode": status_code,
+        "body": body,
+    }
+
+
+def s3_event_handler(event):
+    """Handler function for S3 events.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    # Retrieve the Object name
+    s3_filepath = event["detail"]["object"]["key"]
+
+    filename = os.path.basename(s3_filepath)
+    # TODO: add checks for SPICE or other
+    # data types
+
+    try:
+        science_file = ScienceFilepathManager(filename)
+    except InvalidScienceFileError as e:
+        logger.info(str(e))
+        return http_response(status_code=400, body=str(e))
+
+    # setup a dictionary of metadata parameters to unpack in the
+    # file catalog table
+    metadata_params = science_file.get_file_metadata_params()
+    metadata_params["file_path"] = s3_filepath
+
+    if metadata_params["data_level"] != "l0":
+        logger.info("Invalid data level. Expected l0")
+        return http_response(status_code=400, body="Invalid data level")
```

Comment:
Where does this response end up? Does this somehow show in the upload API?

Reply:
This doesn't get end up in API, I wrote that as default behavior so that I can check in test and also down the road, if we decide to add endpoint or something, have some response will allow checking for error from this lambda.